### DISCUSSION TITLE
Parameter 'ResourceNames' Cannot be found.

### DISCUSSION THREAD

**Discussion by Gbl13** [Azure/ipam#253] [Q&A]
Hello :)
I'm trying to deploy Azure ipam to an existent Azure Infrastructure. 
So i created the $ResourceNames = @..... pointing to my own resources, except by resources that ipam deploys. But when i run the: ./deploy.ps1 `
  -ResourceNames $ResourceNames `
  -Location "brazilsouth"
  
  i get the: "Parameter 'ResourceNames' Cannot be found." could you guys help me with this error?

**Comment by DCMattyG** [comment]
Hi @Gbl13, happy to help here!

First an foremost, at this point today the Azure IPAM deployment cannot reuse existing resources. The `deploy.ps1` script will deploy dedicated Azure infrastructure for the Azure IPAM services. I wasn't sure if you mean by "pointing to my own resources" if you were implying the resources currently exist, or if you were just providing custom resource names, which is totally fine.

I'm sure you've seen our deployment guide, which can be found here:

https://azure.github.io/ipam/#/deployment/README?id=full-deployment

There's an example titled **Override default resource names with custom resource names** which I believe is what you're using as a reference for the above.

For additional troubleshooting, can you help me understand when the last time you pulled the code from GitHub?
- Did you Git Clone the repo, or did you download the ZIP file from GitHub?
- Have you verified that you met all of the prerequisites that are listed in the Deployment section, especially the PowerShell version?
  - https://azure.github.io/ipam/#/deployment/README?id=prerequisites

Always happy to help, just want to make sure we've gotten all of the basics out of the way!

**Reply by Gbl13** [reply]
Wow, thanks for your quick response! 
I was trying to use an existent infrastructure but i dont think it's a problem, i will deploy all the resources and move the webapp to our app service plan. 

To deploy i'm using default azure portal cli powershell (7.4.1 version) and i used the comand git clone https://github.com/Azure/ipam.git. 

Should i use powershell on my computer instead of using on portal?

**Reply by DCMattyG** [reply]
Yes, you need to have all of the [requirements](https://azure.github.io/ipam/#/deployment/README?id=prerequisites) installed including the [Azure PowerShell SDK](https://learn.microsoft.com/en-us/powershell/azure/what-is-azure-powershell) as well as the [Microsoft Graph SDK](https://learn.microsoft.com/en-us/graph/sdks/sdk-installation) as outlined in the list of prerequisites list or the `deploy.ps1` script won't run due to the **#Requires** statements at the top of the script.

I've personally never tested this in [Azure Cloud Shell](https://learn.microsoft.com/en-us/azure/cloud-shell/get-started/classic) so I can't attest if it would work there or not. That said it **should** assuming all of the prerequisite modules are present.
